### DISCUSSION TITLE
chap 11 질문 정리

### DISCUSSION THREAD

**Discussion by inseo24** [JS-OS-study/hyokyung-and-students#36] [General]
Disk Management and Scheduling

**Comment by inseo24** [comment]
<h3>느낀 점 & 궁금한 점 </h3>

- 디스크 매니지먼트 과정을 세세하게 배워서 좋았다.
- 내용 자체는 크게 어렵지 않았으나 그만큼 생략하고 배운 느낌?
- 끝까지 수고하셨씁니다!

<hr />
<h3>문제</h3>

Q. ___은 파일 시스템으로도 쓸 수 있고 swap area로도 쓸 수 있습니다.
___을 파일 시스템으로 만드는 것을 ___ _______이라고 합니다.
A. ___, ___ _______

**Reply by 1chz** [reply]
- Disk
- Partition, Logical Formatting

**Reply by pullingoff** [reply]
Disk, 파티션을 파일시스템으로 만드는 것을 논리 포매팅이라고 합니다.

**Reply by sihyung92** [reply]
Disk, partition, logical formatting

**Comment by 1chz** [comment]
# 느낀점

---

- 디스크 스케쥴링 알고리즘을 죄다 엘리베이터에 대입하며 강의를 들으니 이 알고리즘이 좋다, 나쁘다가 직관적으로 느껴졌다
- 강의가 나온지 좀 돼서 HDD에 대한 내용들이라, SSD를 사용하는 시스템에서는 어떤지 따로 공부해봐야겠다는 생각이 들었다

<br />

# 문제

---

- 이 디스크 스케쥴링 알고리즘은 현대 컴퓨팅 시스템에서 가장 많이 알고리즘 중 하나이며, 이 알고리즘에서 파생된 많은 알고리즘이 다시 현대 컴퓨팅 시스템에서 사용되고 있다. 우리가 사용하는 엘리베이터의 동작과 동일하여 엘리베이터 알고리즘이라고도 불리는 이 알고리즘은 무엇인가?

**Reply by pullingoff** [reply]
오 진짜 SSD에선 어떤지 공부해볼 필요가 있겠네요!

문제 답: SCAN 디스크 스케쥴링 알고리즘

**Reply by inseo24** [reply]
SCAN 스케줄링

**Reply by sihyung92** [reply]
SCAN 알고리즘

**Comment by pullingoff** [comment]
# 느낀 점

- 디스크 스케줄링 알고리즘들을 엘레베이터에 비유해서 배우니 이해가 편했다. 
- 디스크 헤드가 이동을 많이 하면 물리적으로도 데미지가 커서 디스크의 수명이 더 짧을 것 같다.

# 문제
1. (논리/물리) 메모리 시스템에서는 디스크를 메모리의 연장 공간으로 사용할 수 있다.
2. 디스크를 Swap area로 사용할 때 ___ 효율성이 ___ 효율성보다 우선이다.
3. 파티셔닝을 하면 운영체제는 각각의 실린더 그룹을 (별개의/하나의) 디스크로 취급한다.

**Reply by inseo24** [reply]
1. 논리!
2. 속도 효율성이 공간 효율성보다 중요 
3. 별개의

**Reply by sihyung92** [reply]
1. 물리
2. 시간 효율성이 공간 효율성보다 우선이다
3. 별개의 (독립적)

**Reply by 1chz** [reply]
1. 논리
2. 속도, 공간
3. 별개의

**Comment by sihyung92** [comment]
# 느낀 점
- 알고리즘 하면 FCFS가 항상 처음 나오니 ㅎㅎ 반가운 느낌도 든다. 그리고 가장 단순한 형태에서 시작해서, 거기서 문제점을 극복해나가는 식으로 코딩을 하는 자세도 중요하다는 생각도 들고?
- SSD도 전자 기반이라고 들었는데, 메모리와는 어떤 차이가 있어서 더 느릴까?
# 문제
디스크 Access time 의 구성요소를 빠른 순서(Seek Time, Rotational latency, Transfer time)대로 나열해주세요

**Reply by pullingoff** [reply]
rotational latency, transfer time, seek time
!??!

**Reply by inseo24** [reply]
transfer time -> rotational latency -> seek time

**Reply by sihyung92** [reply]
transfer time -> Rotational latency -> Seek Time

**Reply by 1chz** [reply]
transfer time -> rotational latency -> seek time
